src/soft.rs

```diff
@@ -393,6 +394,42 @@ impl Tpm for SoftTpm {
         })
     }
 
+    fn identity_key_decrypt(
+        &mut self,
+        key: &IdentityKey,
+        input: &[u8],
+    ) -> Result<Vec<u8>, TpmError> {
+        let mut decrypter = match key {
+            IdentityKey::SoftEcdsa256 { pkey, x509: _ }
+            | IdentityKey::SoftRsa2048 { pkey, x509: _ } => {
+                Decrypter::new(pkey).map_err(|ossl_err| {
+                    error!(?ossl_err);
+                    TpmError::IdentityKeyInvalidForDecrypting
+                })?
+            }
+        };
+
+        decrypter
```

Comment:
You only need this on RSA, not ECDSA. 

There are also a few styles of RSA padding, https://docs.rs/openssl/latest/openssl/rsa/struct.Padding.html which may be important here especially since in the JWE it specifically calls out RSA-OAEP which likely indicates your padding mode should be https://docs.rs/openssl/latest/openssl/rsa/struct.Padding.html#associatedconstant.PKCS1_OAEP. I'm not sure what the function signature should look like for that, since some key types don't need padding, some do. Maybe we just make our own Padding enum that maps the various options in the openssl version?

Reply:
Added this to testing, and it fails for ecdsa with "operation not supported for this keytype".

Reply:
It's failing initializing the Encrypter.